samples/java-eventsourced-counter-pojo/src/main/java/org/example/domain/CounterState.java

```diff
@@ -0,0 +1,29 @@
+package org.example.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class CounterState {
+
+  public final int value;
+
+  @JsonCreator
+  public CounterState(@JsonProperty("value") int value) {
+    this.value = value;
+  }
+  
+  public static CounterState defaultInstance() {
+    return new CounterState(0);
+  }
+
+  public CounterState increase ( int value){
+    if (value <= 0) throw new IllegalArgumentException("decrease value must be a positive number");
+    return new CounterState(this.value + value);
+  }
+
+  public CounterState decrease ( int value){
+    if (value >= 0) throw new IllegalArgumentException("decrease value must be a negative number");
+    return new CounterState(this.value + value);
```

Comment:
`- value` ?

Reply:
`+ value` is correct. The passed value must be a negative value so `this.value + (- value)`. Either that, or the other way around.

Reply:
strange, then I would have expected increase(-5) instead of having both increase and decrease

Reply:
Yeah, there are a few different ways to implement it. I just followed what we have done in the original sample.